fcrepo-http-api/src/main/java/org/fcrepo/http/api/ContentExposingResource.java

```diff
@@ -847,6 +851,17 @@ void evaluateRequestPreconditions(final Request request,
             throw new PreconditionException(message != null ? message.toString()
                     : "Request failed due to unspecified failed precondition.", response.getStatus());
         }
+
+        final String method = request.getMethod();
+        if (method.equals(HttpPut.METHOD_NAME) || method.equals(HttpPatch.METHOD_NAME)) {
+            final String stateToken = resource.getStateToken();
+            final String clientSuppliedStateToken = headers.getHeaderString("X-If-State-Token");
+            if (clientSuppliedStateToken != null && !stateToken.equals(clientSuppliedStateToken)) {
+                throw new PreconditionException(format(
+                    "The client-supplied value ({0}) does not match the resource's current state token ({1}).",
```

Comment:
[MessageFormat](https://docs.oracle.com/javase/8/docs/api/java/text/MessageFormat.html) has issues with single quotes. The following format produces:
```
The client-supplied value (junk) does not match the resource's current state token ({1}).
```
Instead of:
```
The client-supplied value (junk) does not match the resource's current state token (b644dca3c7af172cf56625bcd5877c247a213a92).
```

Suggestion: Change the message to (note, no singe quote `resource's`):
```
"The client-supplied value ({0}) does not match the current state token ({1})."
```